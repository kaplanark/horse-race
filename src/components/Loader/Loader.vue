<script setup>
const props = defineProps({
	status: {
		type: Boolean,
		default: false
	}
});
</script>

<template>
	<div class="loader" :class="{ 'loader--active': props.status }">
		<div class="loader__content">
			<img src="@assets/images/horse.gif" alt="" width="100">
		</div>
	</div>
</template>

<style lang="scss">
.loader {
	height: 100px;
	width: 100px;
	border-radius: 50%;
	display: flex;
	align-items: center;
	justify-content: center;
	position: relative;

	&::before {
		content: '';
		height: 100%;
		width: 100%;
		border-radius: 50%;
		position: absolute;
		border-top: 2px solid #222226;
		animation: counterclockwise 2.5s linear infinite;
		animation-play-state: var(--loader-animation);
	}

	&::after {
		content: '';
		height: 100%;
		width: 100%;
		border-radius: 50%;
		position: absolute;
		border-bottom: 2px solid #222226;
		animation: clockwise 1.5s linear infinite;
		animation-play-state: var(--loader-animation);
	}

	&--active {
		--loader-animation: running;
	}
}
</style>