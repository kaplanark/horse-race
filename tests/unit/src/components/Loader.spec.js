import { describe, expect, test } from 'vitest'
import { mount } from '@vue/test-utils'
import Loader from '@components/Loader/Loader.vue';

describe('Loader.vue', () => {
	let wrapper;
	beforeEach(() => {
		wrapper = mount(Loader, {
			props: {
				status: true,
			}
		});
	});

	afterEach(() => {
		wrapper.unmount();
	});

	test('is correctly rendered', () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	test('should renders is page content is correct', () => {
		expect(wrapper.vm).toBeTruthy();
	});
	test('renders props.status when passed', () => {
		expect(wrapper.classes()).toContain('loader--active');
	});
});