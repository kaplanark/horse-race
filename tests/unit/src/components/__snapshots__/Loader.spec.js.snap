// Vitest Snapshot v1

exports[`Loader.vue > is correctly rendered 1`] = `
"<div class=\\"loader loader--active\\">
  <div class=\\"loader__content\\">🐎</div>
</div>"
`;
